### PR TITLE
[MM-38421][MM-38507][MM-38676][MM-38772][MM-38876] - Address style bugs for 6.1 release

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
This PR addresses several styling issues for 6.1 release

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38876
https://mattermost.atlassian.net/browse/MM-38772
https://mattermost.atlassian.net/browse/MM-38676
https://mattermost.atlassian.net/browse/MM-38507
https://mattermost.atlassian.net/browse/MM-38421

#### Release Note

-->
```release-note
NONE
```

### CODE REVIEW

File: components/widgets/menu/menu_items/submenu_item.test.tsx [hunk 1]
```diff
@@ -23,13 +21,7 @@ describe('components/widgets/menu/menu_items/submenu_item', () => {
...
         );
 
-        expect(wrapper.html()).toEqual(format(`<li class="SubMenuItem MenuItem" role="menuitem" id="1_menuitem">
-                <div class="" id="1" tabindex="0"><span class="MenuItem__primary-text">test
-                    <span id="channelHeaderDropdownIconRight_1" class="fa fa-angle-right SubMenu__icon-right-empty" aria-label="submenu icon"></span></span>
-                    <ul class="a11y__popup Menu dropdown-menu SubMenu" style="visibility: hidden; right: 100%;"></ul>
-                </div>
-            </li>`),
-        );
+        expect(wrapper.html()).toMatchSnapshot();
```
Comment (hmhealey) [inline components/widgets/menu/menu_items/submenu_item.test.tsx:24]: If we're removing the inline snapshots here, should we also change the `wrapper.html()` back to just `wrapper` then?

**Review by hmhealey — CHANGES_REQUESTED** [review]

**Comment by jgilliam17** [comment]
@nevyangelova Can you please enable Global Header on the test server? Thanks

**Comment by nevyangelova** [comment]
@jgilliam17 its on for me.

**Comment by jgilliam17** [comment]
Thanks @nevyangelova It's on for me too today
Seeing few issues 
1. Menu is not centered in mobile webview
2. Separator line theming  is off 
3. Submenu arrow icon is taking an extra row, not sure if this intended in mobile view

4. Out of scope. On Chrome (not on Firefox) seeing here on the test PR and also on daily - we surface `...` as if we are truncating menu option text. Not sure why that is happening.
Screen Shot 2021-10-05 at 10 34 17 AM

**Comment by nevyangelova** [comment]
@jgilliam17 kind reminder to review

**Review by michelengelen — APPROVED** [review]
LGTM

**Review by michelengelen — APPROVED** [review]

**Review by michelengelen — CHANGES_REQUESTED** [review]
@nevyangelova it seems as if a lot of changes from #9123 came in as changes ... could this be due to a bad merge?

**Comment by michelengelen** [comment]
@nevyangelova nevermind my last comment ... it seems github did not update the changes correctly!

**Review by michelengelen — COMMENTED** [review]
one additional thing that came to mind: why are you switching between `flex` and `grid` it should be doable with only `grid` IMO.

**Review by michelengelen — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @nevyangelova 
Tested, looks good to merge.
- Verified ‘move to’ in mobile, spacing on menus, username opacity and mention & saved tooltips with RHS open. 
Note: issue with username opacity on dark theme will be fixed after compass changes per [discussion here](https://github.com/mattermost/mattermost-webapp/pull/9002#issuecomment-945890581)
Nevy - let me know if we already have a ticket for this, if not I can create one. Thanks.

**Review by hmhealey — APPROVED** [review]

**Pull request merged**